digits_test.go

```diff
@@ -7,250 +7,334 @@ import (
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "396006000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "45121000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "45121",
+			nonSigFigs: "000",
+			strOut:     "45,121,000.00",
+			head:       "",
+			core:       "45,121",
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "487000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "98000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "98",
+			nonSigFigs: "000",
+			strOut:     "98,000.00",
+			head:       "",
+			core:       "98",
+			tail:       ",000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "4059000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Thousands,
+			value:               "-338863000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "338863",
+			nonSigFigs: "000",
+			strOut:     "(338,863,000.00)",
+			head:       "(",
+			core:       "338,863",
+			tail:       ",000.00)",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "45121000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.Exact,
+			value:               "1038807",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "1038807",
+			nonSigFigs: "",
+			strOut:     "1,038,807.00",
+			head:       "",
+			core:       "1,038,807.",
+			tail:       "00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "98000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.HundredThousands,
+			value:               "80800000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "808",
+			nonSigFigs: "00000",
+			strOut:     "80,800,000.00",
+			head:       "",
+			core:       "80,8",
+			tail:       "00,000.00",
+		},
 	},
 	{
-		p: digits.Thousands,
-		v: "-338863000",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
+			precision:           digits.HundredThousands,
+			value:               "-80800000",
+			groupSeparator:      ',',
+			fractionalPrecision: digits.PreserveUpToHundredth,
+		},
+		output: output{
+			sigFigs:    "808",
+			nonSigFigs: "00000",
+			strOut:     "(80,800,000.00)",
+			head:       "(",
+			core:       "80,8",
+			tail:       "00,000.00)",
+		},
 	},
 	{
-		p: digits.Exact,
-		v: "1038807",
-		g: ',',
-		d: digits.PreserveUpToHundredth,
+		stimulus: stimulus{
```

Comment:
This is a good addition to the test coverage, but I would declare this out of scope of the current work item.

Comment out this test case with a TODO and a url to the reported bug like this:

```

	// {	//todo BUG <url to github issue here>
	// 	stimulus: stimulus{
	// 		precision:           digits.Tenth,
	// 		value:               "80800000.99090909090",
	// 		groupSeparator:      ',',
	// 		fractionalPrecision: digits.PreserveUpToHundredth,
	// 	},
	// 	output: output{
	// 		sigFigs:    "80800000.9",
	// 		nonSigFigs: "9",
	// 		strOut:     "80,800,000.99",
	// 		head:       "",
	// 		core:       "80,800,000.9",
	// 		tail:       "9",
	// 	},
	// },
```

Reply:
#7

Reply:
I'm going to clarify why this is out of scope.  Currently `telefacts` (an XBRL processor) is the major consumer of `digits`.  A numeric expression like this practically will never ever happen within the XBRL community.  This is a numeric value that mathematicians and scientists would come up with.  Therefore, fixing this is not a priority as this will never be seen as an XBRL numeric fact.